### PR TITLE
Emails encoding

### PR DESCRIPTION
**Pull request by Glomberg**

### CODE REVIEW

File: lib/Cleantalk/Antispam/EmailEncoder.php [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        add_action('wp_ajax_nopriv_apbct_decode_email', array($this, 'ajaxDecodeEmailHandler'));
+        add_action('wp_ajax_apbct_decode_email', array($this, 'ajaxDecodeEmailHandler'));
+    }
+
+    public function modifyContent($content)
+    {
+        global $apbct;
+
+        return preg_replace_callback('/([_A-Za-z0-9-]+(\.[_A-Za-z0-9-]+)*@[A-Za-z0-9-]+(\.[A-Za-z0-9-]+)*(\.[A-Za-z]{2,}))/', function ($matches) {
+            if ( in_array(strtolower($matches[4]), ['.jpg', '.jpeg', '.png', '.gif', '.svg']) ) {
```
Comment (alexandergull) [inline lib/Cleantalk/Antispam/EmailEncoder.php:54]: .webp can also persist

**Review by alexandergull — CHANGES_REQUESTED** [review]

File: cleantalk.php [hunk 2]
```diff
@@ -141,6 +141,14 @@
     \Cleantalk\Antispam\DisableComments::getInstance();
 }
 
+// Email encoder
+if (
+    $apbct->key_is_ok &&
```
Comment (ArtemAnoshin) [inline cleantalk.php:146]: what do you think if you replace such multi-storey conditions with a single function like apbct_is_encoder_on()
You can even create a separate file under these conditions, we have a lot of them

File: js/src/apbct-public.js [hunk 3]
```diff
@@ -268,11 +268,74 @@
...
+			}
+		}
 	}
 	apbct_attach_event_handler(window, "DOMContentLoaded", apbct_ready);
 
 }());
 
+function apbctAjaxEmailDecode(event){
+	const element = event.target;
+	element.setAttribute('title', 'Wait for decoding...');
```
Comment (ArtemAnoshin) [inline js/src/apbct-public.js:296]: the text will be without translation

File: cleantalk.php [hunk 4]
```diff
@@ -141,6 +141,14 @@
...
+    ( ! is_admin() || apbct_is_ajax() ) &&
+    $apbct->settings['data__email_decoder'] ) {
+    \Cleantalk\Antispam\EmailEncoder::getInstance();
```
Comment (ArtemAnoshin) [inline cleantalk.php:149]: why such a construction, the object is not used further...

Comment (Glomberg) [reply cleantalk.php:149]: The instance is used into the class code by $this link

File: cleantalk.php (hunk 2 above)
Comment (Glomberg) [reply cleantalk.php:146]: Maybe but not sure

File: lib/Cleantalk/Antispam/EmailEncoder.php (hunk 1 above)
Comment (Glomberg) [reply lib/Cleantalk/Antispam/EmailEncoder.php]: Done https://github.com/CleanTalk/wordpress-antispam/commit/f532f01953e3f86a31742b8fbbd4f1bea7595d39

File: js/src/apbct-public.js (hunk 3 above)
Comment (Glomberg) [reply js/src/apbct-public.js]: Done https://github.com/CleanTalk/wordpress-antispam/commit/79013d756cb0ee12e42616f84c0655c98f441e08

**Review by alexandergull — APPROVED** [review]

**Pull request merged**